ui/raidboss/data/06-ew/trial/zodiark-ex.ts

```diff
@@ -19,6 +192,401 @@ const triggerSet: TriggerSet<Data> = {
+            data.activeSigils.push({ x: actor.PosX, y: actor.PosY, typeId: '67E5', npcId: actor.ID.toString(16) });
+          }
+        }
+      },
+      alertText: (data, matches, output) => {
+        const target = data.activeSigils[data.activeSigils.length - 1];
+        if (!target) {
+          console.error(`Red Tether: Missing target, ${JSON.stringify(matches)}`);
+          return;
+        }
+
+        if (target.x < 100)
+          return output.east!();
+
+        if (target.x > 100)
+          return output.west!();
+
+        if (target.y < 100)
+          return output.south!();
+        return output.north!();
+      },
+      outputStrings: directionOutputStrings,
+    },
+    {
+      id: 'ZodiarkEx Roiling Darkness Spawn',
+      type: 'AddedCombatant',
+      netRegex: NetRegexes.addedCombatant({ name: 'Roiling Darkness', capture: false }),
+      suppressSeconds: 1,
+      response: Responses.killAdds(),
+    },
+    {
+      // 67E4 Green Beam, 67E5 Rectangle, 67E6 Wedge
+      id: 'ZodiarkEx Arcane Sigil Start',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['67E4', '67E5', '67E6'], source: 'Arcane Sigil' }),
+      run: (data, matches, _output) => {
+        if (parseFloat(matches.y) < 100)
+          data.activeFrontSigils.push({ x: parseFloat(matches.x), y: parseFloat(matches.y), typeId: matches.id, npcId: matches.sourceId });
+      },
+    },
+    {
+      id: 'ZodiarkEx Arcane Sigil Start Cleanup',
+      type: 'StartsUsing',
+      netRegex: NetRegexes.startsUsing({ id: ['67E4', '67E5', '67E6'], source: 'Arcane Sigil', capture: false }),
+      delaySeconds: 0.2,
+      suppressSeconds: 0.2,
+      alertText: (data, _matches, output) => {
+        const activeFrontSigils = data.activeFrontSigils;
+        data.activeFrontSigils = [];
+        if (activeFrontSigils.length === 1 && activeFrontSigils[0]?.typeId === '67E4')
+          return output.sides!();
+        if (activeFrontSigils.length === 2 && activeFrontSigils[0]?.typeId === '67E4' && activeFrontSigils[1]?.typeId === '67E4')
+          return output.middle!();
+        if (activeFrontSigils.length === 3) {
+          for (const sig of activeFrontSigils) {
+            // Find the middle sigil
+            if (sig.x > 90 && sig.x < 110) {
+              if (sig.typeId === '67E4')
+                return output.frontsides!();
+              if (sig.typeId === '67E5')
+                return output.backmiddle!();
+              if (sig.typeId === '67E6')
+                return output.frontmiddle!();
+            }
+          }
+        }
+      },
+      outputStrings: {
+        frontsides: {
+          en: 'front sides',
+        },
+        backmiddle: {
+          en: 'back middle',
+        },
+        frontmiddle: {
+          en: 'front middle',
+        },
+        sides: Outputs.sides,
+        middle: Outputs.middle,
```

Comment:
True, the output is a big TODO, unfortunately the star callouts are extremely late anyways because the StartsCasting for the meteors is very late.

I don't really see how I could get this information earlier. (maybe actually detect the star patterns somehow?)

If I don't get an epiphany the way to go would likely be to remove star callouts.

I agree that middle first or middle then sides would be a much better callout, this is also possible with current technology so no big problem there.

Reply:
Yeah, I'm sorta torn.  It's slightly better than nothing, in that if you're running to north and it says middle, you could possibly course correct with sprint up, but it'd be tight.  But maybe it's better to just not have anything so that people don't attempt to rely on it.

Yeah, I think you'd have to detect the star pattern, but I don't think it's in the log.  Alternatively it's possible that the combatants are placed prior to starts casting and you could detect them earlier with a well-timed getCombatants call.